scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
+      * @param lengthFieldOffset the offset of the length field.
+      * @param lengthFieldLength  the length of the length field. Must be 1, 2, 3, 4 or 8 bytes.
+      * @param encodingLengthAdjustment the compensation value to add to the value of the length field when encoding.
+      * @param decodingLengthAdjustment the compensation value to add to the value of the length field when decoding.
+      * @param initialBytesToStrip the number of first bytes to strip out from the decoded frame. In standard framing setup
+      *                            it should be equal to lengthFieldLength.
+      * @param failFast  If true, a DecodingError event is generated as soon as the decoder notices the length of the frame will
+      *                  exceed maxFrameLength regardless of whether the entire frame has been read. If false,
+      *                  a a DecodingError event is generated after the entire frame that exceeds maxFrameLength has been read..
+      * @param byteOrder the ByteOrder of the length field.
+      * @param lengthIncludesLengthFieldLength if true, the length of the prepended length field is added to the value of the prepended length field.
+      */
+    def buildConfig(
+        maxFrameLength: Int,
+        lengthFieldOffset: Int,
+        lengthFieldLength: Int,
+        encodingLengthAdjustment: Int,
+        decodingLengthAdjustment: Int,
+        initialBytesToStrip: Int,
+        failFast: Boolean = true,
+        byteOrder: ByteOrder = ByteOrder.BIG_ENDIAN,
+        lengthIncludesLengthFieldLength: Boolean = false
+    ): Option[FramingConfig] = {
```

Comment:
I wonder if we really have to expose all these options, rather than just sensible defaults. I understand they are added because Netty supports them, but what if we decided not to use Netty for this after all, and these settings would not map clearly to whatever replaces it?

Reply:
My take on that is that at this point, it is more probable that we will fiddle with framing format than change natty itself, therefore I thought to have one constructor that expose everything (in case it will be needed) and second constructor (`buildConfigWithStrippedLength`) with sensible defaults with framing format which we use right now.

Reply:
The only reason I can imagine you wanting to change it is if the other side had some obscure format that doesn't match the defaults. I think in at least some codecs we had: `length-of-length-of-content || length-of-content || content`, so the actual length prefix was dynamic. I'm not sure if these config values could capture that, since while you know the offset of the length-prefix, it doesn't have a fixed length. 

Fine though, it's no problem to ignore these. I agree we're not getting rid of Netty; it's more likely people would rather want to switch away from Scalanet to https://github.com/typelevel/fs2-netty than change Netty.

Reply:
oh nice i did not know such projects existed, it is definitly worth keeping an eye on it